### PR TITLE
Made --failed-tests actually import only failed tests 

### LINKED ISSUE #22572: run_tests param --failed-tests does not actually run only failed tests
## Bug Description
Depending on how tests are ran, the previous_test_results.json file generated will not properly mark all tests as successes or silent, and keeps many as pending. Due to the way the failed-tests option works, it gathers all tests that are pending as well, rather than just those that fail.

## Steps to Reproduce
Easiest way to trigger this problem is by hitting the Max Failure Count. 

1. Go into a folder with multiple tests, edit one test so it fails
2. Run ./run_tests --max-fails 1 --re "folder name" -j {however many jobs}
3. After testing ends, run ./run_tests -j {however many jobs} --failed-tests

You will see all the jobs in that folder run.

## Impact
For most, an annoyance, but can be extremely difficult to deal with when trying to, say, debug things with a large failure count in a large folder like modules, for example. Makes it impossible to properly utilize failed-tests in these cases

### PR DESCRIPTION
**Pull request by socratesgorilla**
Closes #22572 

Definitely one of those changes that is simple once you find where the problem actually lies, but finding where the problem lies is the difficult part.

### CODE REVIEW

File: python/TestHarness/JobDAG.py [hunk 1]
```diff
@@ -166,8 +166,8 @@ def _doPreviouslyFailed(self, job):
             tester.addCaveats('re-running')
             return
 
-        # This job was skipped, passed or silent
-        elif status in job.job_status.getSuccessStatuses():
+        # This job was skipped, passed, pending or silent
+        elif status in job.job_status.getSuccessStatuses()+job.job_status.getPendingStatuses():
```
Comment (loganharbour) [inline python/TestHarness/JobDAG.py:170]:
```suggestion
        elif status in job.job_status.getSuccessStatuses() or status in job.job_status.getPendingStatuses():
```

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Comment by GiudGiud** [comment]
I actually kind of like the current behavior where --failed-tests gets all the failed ones + all the ones that did not run yet. That way you can restart the test suite where you left it after fixing something

**Comment by GiudGiud** [comment]
I ll live either way

**Comment by socratesgorilla** [comment]
> I actually kind of like the current behavior where --failed-tests gets all the failed ones + all the ones that did not run yet. That way you can restart the test suite where you left it after fixing something

Want me to add a new param for old behavior?

**Comment by GiudGiud** [comment]
nah it s fine thanks though

File: python/TestHarness/JobDAG.py (hunk 1 above)
Comment (socratesgorilla) [reply python/TestHarness/JobDAG.py:170]: Is there a meaningful difference to this? They both work and list concat is super quick in python, so I'm not even sure the suggestion is faster.

Comment (socratesgorilla) [reply python/TestHarness/JobDAG.py:170]: Also important to note that these are just getting the enums for the error codes, and the total list is like size 8. Pending statuses is a 3 size list and success statuses is like 5 or 6. Obviously the list concat is bad practice on a larger/dynamic list but here I think its fine and appropriate

Comment (socratesgorilla) [reply python/TestHarness/JobDAG.py:170]: Important to note that I also originally wrote it the way the commit suggests, then realized there was an equally pythonic way to do it that would also look cleaner and noticed no performance loss from the change.

Comment (loganharbour) [reply python/TestHarness/JobDAG.py:170]: It's just a preference; avoids construction of temporaries and doesn't read any better/worse

**Comment by socratesgorilla** [comment]
@GiudGiud @loganharbour can this get merged?

**Review by GiudGiud — CHANGES_REQUESTED** [review]
let s squash all this in a single commit

**Review by GiudGiud — APPROVED** [review]

**Comment by socratesgorilla** [comment]
@GiudGiud @loganharbour good?

**Review by loganharbour — APPROVED** [review]
Ditto - re-running tests first

**Comment by GiudGiud** [comment]
Tests passed

**Pull request merged**